versioned/mongodb/src/main/java/com/dremio/nessie/versioned/store/mongodb/MongoDbStore.java

```diff
@@ -0,0 +1,200 @@
+    this.serverName = serverName;
+    this.databaseName = databaseName;
+    this.pojoCodecRegistry = fromRegistries(MongoClientSettings.getDefaultCodecRegistry(),
+      fromProviders(PojoCodecProvider.builder().automatic(true).build()));
+    this.mongoClientSettings = MongoClientSettings.builder()
+      .applyToClusterSettings(builder ->
+        builder.hosts(Arrays.asList(new ServerAddress(serverName, mongoPort))))
+      .codecRegistry(pojoCodecRegistry)
+      .build();
+  }
+
+  /**
+   * Gets a handle to an existing database or get a handle to a MongoDatabase instance if it does not exist. The new database will be lazily created.
+   * Since MongoDB creates databases and collections if they do not exist, there is no need to validate the presence of
+   * either before they are used. This creates or retrieves references collections that map 1:1 to the enumerates in
+   * {@link com.dremio.nessie.versioned.store.ValueType}
+   */
+  @Override
+  public void start() {
+    try (MongoClient mongoClient = MongoClients.create(mongoClientSettings)) {
+      mongoDatabase = mongoClient.getDatabase(databaseName);
+    }
+    //Initialise collections for each ValueType.
+    l1MongoCollection = mongoDatabase.getCollection(l1Collection, L1.class);
+    l2MongoCollection = mongoDatabase.getCollection(l2Collection, L2.class);
+    l3MongoCollection = mongoDatabase.getCollection(l3Collection, L3.class);
+    MongoCollection<InternalRef> refMongoCollection = mongoDatabase.getCollection(refCollection, InternalRef.class);
+    valueMongoCollection = mongoDatabase.getCollection(valueCollection, InternalValue.class);
+    MongoCollection<Fragment> keyFragmentMongoCollection = mongoDatabase.getCollection(keyFragmentCollection, Fragment.class);
+    MongoCollection<InternalCommitMetadata> commitMetadataMongoCollection =
+        mongoDatabase.getCollection(commitMetadataCollection, InternalCommitMetadata.class);
+  }
+
+  /**
+   * Closes the connection this manager creates to a database. If the connection is already closed this method has
+   * no effect.
+   */
+  @Override
+  public void close() {
+  }
+
+  @Override
+  public void load(LoadStep loadstep) throws ReferenceNotFoundException {
+    throw new UnsupportedOperationException();
+  }
+
+  private List<ListMultimap<String, LoadOp<?>>> paginateLoads(LoadStep loadStep, int size) {
+    List<ListMultimap<String, LoadOp<?>>> paginated = new ArrayList<>();
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public <V> boolean putIfAbsent(ValueType type, V value) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  @SuppressWarnings("unchecked")
+  public <V> void put(ValueType type, V value, Optional<ConditionExpression> conditionUnAliased) {
+    // TODO ensure that calls to mongoDatabase.createCollection etc are surrounded with try-catch to detect
+    //  com.mongodb.MongoSocketOpenException
+    LOGGER.info("Connected to ", mongoDatabase.toString());
```

Comment:
slf4j supports formatted strings, so string concatenation should never be used. I'll add a checker for this.

Reply:
Good that I exposed that :) I'll use formatted. Was just being lazy.

Reply:
String concatenation has been removed.